contracts/proposal/dao-proposal-single/src/contract.rs

```diff
@@ -529,15 +559,36 @@ pub fn execute_vote(
             }
         }
         None => Ok(Ballot {
-            power: vote_power,
+            power: vote_power.total,
+            individual_power: vote_power.individual,
             vote,
             rationale: rationale.clone(),
         }),
     })?;
 
+    // DELEGATE VOTE OVERRIDE: if this is the first time this member voted,
+    // override their delegates' votes with the delegator's vote.
+    if is_first_vote {
+        handle_delegate_vote_override(
+            deps.branch(),
+            &sender,
+            &prop.delegation_module,
+            &env.contract.address,
+            proposal_id,
+            prop.start_height,
+            &vote_power.individual,
+            BALLOTS,
+            &mut |vote, power| {
+                prop.votes.remove_vote(*vote, power);
+                Ok(())
+            },
+        )?;
+    }
```

Comment:
I like this design. can't think of a lower lift to separate the concerns between the delegations module and proposals. i think even if something went ultra wrong with delegations, every member voting this way would be all that's needed to assert the final say in that VP, right?

Reply:
yep i think so